src/main/java/org/broadinstitute/hellbender/engine/dataflow/datasources/RefAPISource.java

```diff
@@ -153,4 +198,105 @@ private static Genomics createGenomicsService( final PipelineOptions pipelineOpt
+        Map<String, String> ret = new HashMap<>();
+        for (Map.Entry<String,Reference> e : referenceMap.entrySet()) {
+            ret.put(e.getKey(), e.getValue().getId());
+        }
+        return ret;
+    }
+
+    private SAMSequenceDictionary getReferenceSequenceDictionaryFromMap(final Map<String, Reference> referenceMap, final SAMSequenceDictionary optReadSequenceDictionaryToMatch) {
+        SAMSequenceDictionary refDictionary = new SAMSequenceDictionary();
+        ArrayList<SAMSequenceRecord> refContigs = new ArrayList<>();
+
+        for (Map.Entry<String, Reference> e : referenceMap.entrySet()) {
+            if (e.getKey()!=null && e.getValue().getLength()!=null) {
+                refContigs.add(new SAMSequenceRecord(e.getKey(), e.getValue().getLength().intValue()));
+            }
+        }
+
+        HashMap<String,Integer> indexBuilder = null;
+        if (null!=optReadSequenceDictionaryToMatch) {
+            indexBuilder = new HashMap<>();
+            for (int i=0; i<optReadSequenceDictionaryToMatch.size(); i++) {
+                final SAMSequenceRecord sequence = optReadSequenceDictionaryToMatch.getSequence(i);
+                indexBuilder.put(sequence.getSequenceName(), i);
+            }
+        }
+        final HashMap<String,Integer> optReadSequenceDictionaryToMatchIndex = indexBuilder;
+
+        // GATK requires human contigs in karyotypic order: 1, 2, ..., 10, 11, ..., 20, 21, 22, X, Y with M either leading or trailing these contigs.
+        // So we sort them.
+        Collections.sort(refContigs, new Comparator<SAMSequenceRecord>() {
+            @Override
+            public int compare(SAMSequenceRecord o1, SAMSequenceRecord o2) {
+
+                // if those are ordered in the readDictionary, then match that order
+                if (null!=optReadSequenceDictionaryToMatchIndex) {
+                    if (optReadSequenceDictionaryToMatchIndex.containsKey(o1.getSequenceName()) && optReadSequenceDictionaryToMatchIndex.containsKey(o2.getSequenceName())) {
+                        return optReadSequenceDictionaryToMatchIndex.get(o1.getSequenceName()).compareTo(optReadSequenceDictionaryToMatchIndex.get(o2.getSequenceName()));
+                    }
+                }
+
+                // otherwise, order them in karyotypic order.
+                int r1 = getRank(o1.getSequenceName());
+                int r2 = getRank(o2.getSequenceName());
+                if (r1<r2) return -1;
+                if (r2<r1) return 1;
+                return o1.getSequenceName().compareTo(o2.getSequenceName());
+            }
+
+            private int getRank(String name) {
+                if (name.equalsIgnoreCase("x")) return 23;
+                if (name.equalsIgnoreCase("y")) return 24;
+                if (name.equalsIgnoreCase("m")) return 25;
+                if (name.equalsIgnoreCase("mt")) return 25;
```

Comment:
oh you'll still get a reference dictionary with non-human DNA. GATK only specifically asks for karyotypic order for *human* DNA, so the code does this to please it. Since there is only a requirement for human DNA, the fact that non-human DNA may be sorted differently isn't a problem.